src/function/window/window_merge_sort_tree.cpp

```diff
@@ -26,7 +27,19 @@ WindowMergeSortTree::WindowMergeSortTree(ClientContext &context, const vector<Bo
 	payload_layout.Initialize(payload_types);
 
 	auto &buffer_manager = BufferManager::GetBufferManager(context);
-	global_sort = make_uniq<GlobalSortState>(buffer_manager, orders, payload_layout);
+	if (unique) {
+		vector<BoundOrderByNode> unique_orders;
+		for (const auto &order : orders) {
+			unique_orders.emplace_back(order.Copy());
+		}
+		auto unique_expr = make_uniq<BoundConstantExpression>(Value(index_type));
```

Comment:
So if I understand correctly, you are ordering by `NULL` for `row_number` because it does not need to be ordered?

Reply:
The sort code only needs the type, not the values. This is just a hack to set that up.

Reply:
(We actually do need to order by row number to break ties and produce unique reordered row numbers.)